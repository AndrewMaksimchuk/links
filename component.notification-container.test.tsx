import { test, expect } from 'bun:test'
import { renderToReadableStream } from 'hono/jsx/streaming'
import { NotificationContainer } from './component.notification-container'

test('Component NotificationContainer', async () => {
    const stream = renderToReadableStream(<NotificationContainer />)
    expect(await Bun.readableStreamToText(stream)).toMatchSnapshot()
})