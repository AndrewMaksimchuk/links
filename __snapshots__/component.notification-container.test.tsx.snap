// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Component NotificationContainer 1`] = `
"<section id="notification" style="--pico-background-color:var(--pico-form-element-background-color);--pico-border-color:var(--pico-form-element-border-color);--pico-color:var(--pico-form-element-color);--pico-box-shadow:none;position:fixed;max-width:300px;outline:0;top:var(--pico-form-element-spacing-vertical);right:var(--pico-form-element-spacing-horizontal);border:var(--pico-border-width) solid var(--pico-border-color);border-radius:var(--pico-border-radius);background-color:var(--pico-background-color);box-shadow:var(--pico-box-shadow);color:var(--pico-color);font-weight:var(--pico-font-weight);transition:background-color var(--pico-transition), border-color var(--pico-transition), color var(--pico-transition), box-shadow var(--pico-transition)"><script>
                const onNotify = new CustomEvent('notifyClose', { bubbles: true })
                document.body.addEventListener("notifyClose", function(event){
                    const selector = '[data-notificationid="' + event.detail.notificationid + '"]'
                    setTimeout(() => document.querySelectorAll(selector)[0]?.remove(), 3000)
                })
            </script></section>"
`;
